### ISSUE TITLE
Live editing switch for synchronous editing ability

### ISSUE THREAD

**Issue by ViditChitkara** [publiclab/mapknitter#943]
This is a part of #893 

<img width="677" alt="Screenshot 2019-07-30 at 11 52 45 PM" src="https://user-images.githubusercontent.com/22616017/63088803-98f75100-bf73-11e9-9361-00cb595d5024.png">
A live editing switch is required somewhere on the sidebar. It could be something like:-

![image](https://user-images.githubusercontent.com/24359/63034696-c2d34980-be87-11e9-9fd0-130a11bab0e0.png)
The purpose of this switch is the ability make the map editable by other users connected to the map. 

I've started with #937 . Maybe we could extend that accordingly.

We can use the `toggle_editing` route to save the map creator's preference there.

**Comment by ViditChitkara** [comment]
The sidebar code could be found in https://github.com/publiclab/mapknitter/blob/main/app/views/layouts/knitter2.html.erb file.

**Comment by ViditChitkara** [comment]
We may also take some hints from #381 for the enable/disable apis.

**Comment by jywarren** [comment]
This sounds great. I think we should also have a switch for "allow others
to edit this map" to distinguish from "allow others to edit in real-time"
-- perhaps taking on both these in coordination will help them make sense
in UI design next to one another. Vidit, would you be willing to take on
adding both to ensure they are coherent? (also noting that anonymous maps
will not allow stopping others from editing)

**Comment by ViditChitkara** [comment]
Yes, I think that makes sense. So in total we'll have two switches one for "allow others to edit this map" which will make use of the added column in map table. And the other would simply be "allow real-time-editing". The later would require an extra layer of authentication at connection.rb to stop others from editing the map -- probably by rejecting their connection to the channel -- the same way we do for a user who is not logged in. Does that make any sense?
Also, do you want me to code this or should this remain a first timer only issue?

**Comment by jywarren** [comment]
That sounds great! I think this requires enough coordination that it might
be best if you did this one yourself? Thanks!!

**Comment by ViditChitkara** [comment]
Great!! I'll get started with this.
